extension/opampextension/config.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opampextension
+
+import (
+	"errors"
+
+	"github.com/oklog/ulid/v2"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+// Config has the configuration for the opamp extension.
+type Config struct {
+	confighttp.HTTPClientSettings `mapstructure:",squash"`
```

Comment:
In my opinion we should only expose configuration settings that will be passed to [`Dialer.Dial`](https://pkg.go.dev/github.com/gorilla/websocket#Dialer.Dial). From what I can tell, `confighttp.HTTPClientSettings` is used to configure an HTTP client that is constructed through the `HTTPClientSettings.ToClient` method, though I could be missing some uses.

My overall concern is that several settings exposed in `confighttp.HTTPClientSettings` won't actually affect anything.